RestaurantReservation.Db/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Repositories.IRepositories;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class OrderRepository : EntityRepository<Order>, IOrderRepository
+    {
+
+        public OrderRepository(RestaurantReservationDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public async Task<decimal> CalculateAverageOrderAmountAsync(int employeeId)
+        {
+            var OrderAmountList = await _dbContext.Orders.AsNoTracking()
+                       .Where(o => o.EmployeeId == employeeId)
+                       .Include(o => o.OrderItems)
+                       .ThenInclude(oi=> oi.MenuItem)
+                       .AsSplitQuery()
+                       .ToListAsync();
+
+            return OrderAmountList.Count() > 0 ?
+                 OrderAmountList.Select(o => o.TotalAmount).Average() : 0;
```

Comment:
so you're getting all orders to the server side to return only the average amount for 'em ? is there any other way to do it in more performant way ?